### PR TITLE
feat: add and test actions workflow for pull requests issued to main …

### PR DESCRIPTION
**Pull request by PreetinderSingh345**

### CODE REVIEW

**Comment by PreetinderSingh345** [comment]
Will create another GitHub token to test the workflow

**Comment by PreetinderSingh345** [comment]
The docker image is pushed to the current repository's packages. Therefore permission is needed inside Settings>Actions>General>Workflow Permissions to allow read and write permissions for the current repository. Please allow those permissions, for all the tests to pass.

**Comment by AshreneRoy** [comment]
> The docker image is pushed to the current repository's packages. Therefore permission is needed inside Settings>Actions>General>Workflow Permissions to allow read and write permissions for the current repository. Please allow those permissions, for all the tests to pass.

Looks like I don't have the permission to do this, it rests with Org owners. @Ar11rA @rkdixit96 can you please help out with above for this repo, here's the link to the settings page: https://github.com/tech-university-india/k8s-designer-frontend/settings/actions

File: .github/workflows/buildAndDeploy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build and Deploy
+
+on:
+  pull_request:
+    branches:
+      - main
+env:
+  REGISTRY: ghcr.io
+  USER: preetindersingh345
```
Comment (AshreneRoy) [inline .github/workflows/buildAndDeploy.yml:9]: Do we need user?

File: .github/workflows/buildAndDeploy.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        npm run test
+    - name: Create build
+      run: |
+        npm run build
+    - name: Build Docker image
+      run: |
+        docker build -t ${{env.IMAGE_NAME}} .
+    - name: Login to GHCR
+      run: |
+        echo ${{env.PAT}} | docker login ${{env.REGISTRY}} -u ${{env.USER}} --password-stdin
```
Comment (AshreneRoy) [inline .github/workflows/buildAndDeploy.yml:41]: Let's not echo secrets

Comment (AshreneRoy) [reply .github/workflows/buildAndDeploy.yml:41]: GITHUB_TOKEN is already in env, no need to create another env to store it

File: .github/workflows/buildAndDeploy.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+name: Build and Deploy
+
+on:
+  pull_request:
+    branches:
+      - main
+env:
+  REGISTRY: ghcr.io
+  USER: preetindersingh345
+  IMAGE_NAME: docker-image-name
```
Comment (AshreneRoy) [inline .github/workflows/buildAndDeploy.yml:10]: This should follow repo name

File: .github/workflows/buildAndDeploy.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Create build
+      run: |
+        npm run build
+    - name: Build Docker image
+      run: |
+        docker build -t ${{env.IMAGE_NAME}} .
+    - name: Login to GHCR
+      run: |
+        echo ${{env.PAT}} | docker login ${{env.REGISTRY}} -u ${{env.USER}} --password-stdin
+        docker tag ${{env.IMAGE_NAME}} ${{env.REGISTRY}}/${{env.USER}}/${{env.IMAGE_NAME}}:${{env.TAG}}
```
Comment (AshreneRoy) [inline .github/workflows/buildAndDeploy.yml:42]: Follow naming convention like: 
`<repo_name>:<semVer>-<last_7_digits_of_commit_sha>`

**Review by AshreneRoy — CHANGES_REQUESTED** [review]

File: .github/workflows/buildAndDeploy.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+name: Build and Deploy
```
Comment (AshreneRoy) [inline .github/workflows/buildAndDeploy.yml:1]: We should also test the code base for linting and tests before building and pushing images

Comment (AshreneRoy) [reply .github/workflows/buildAndDeploy.yml:1]: In this workflow, rename the file and keep only the npm run commands for installing packages, checking linting, testing etc.

**Comment by PreetinderSingh345** [comment]
Made the mentioned changes for this branch.

File: .github/workflows/buildAndDeploy.yml (hunk 1 above)
Comment (PreetinderSingh345) [reply .github/workflows/buildAndDeploy.yml]: The image will be pushed to the packages section for the repository and we need to add the user/owner to whom the repository belongs.

**Review by AshreneRoy — APPROVED** [review]

File: .github/workflows/buildAndDeploy.yml (hunk 3 above)
Comment (PreetinderSingh345) [reply .github/workflows/buildAndDeploy.yml]: I'll change it to tech-university-india

File: .github/workflows/buildAndDeploy.yml (hunk 2 above)
Comment (PreetinderSingh345) [reply .github/workflows/buildAndDeploy.yml]: I've added` >/dev/null 2>&1` after the echo statement, it'll suppress the output of the shell script command and the secret will not be visible in the logs.

**Pull request merged**